File: typescript-sdk/src/sdk/models/shared/getrooms.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class GetRoomsData extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=createdAt" })
  createdAt?: string;

  @SpeakeasyMetadata({ data: "json, name=defaultAccesses" })
  defaultAccesses?: any;

  @SpeakeasyMetadata({ data: "json, name=groupsAccesses" })
  groupsAccesses?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=lastConnectionAt" })
  lastConnectionAt?: string;

  @SpeakeasyMetadata({ data: "json, name=metadata" })
  metadata?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type?: string;

  @SpeakeasyMetadata({ data: "json, name=usersAccesses" })
  usersAccesses?: Record<string, any>;
}


export class GetRooms extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=data", elemType: GetRoomsData })
  data?: GetRoomsData[];

  @SpeakeasyMetadata({ data: "json, name=nextPage" })
  nextPage?: string;
}

File: go-sdk/pkg/models/shared/getrooms.go
package shared

type GetRoomsData struct {
	CreatedAt        *string                `json:"createdAt,omitempty"`
	DefaultAccesses  *interface{}           `json:"defaultAccesses,omitempty"`
	GroupsAccesses   map[string]interface{} `json:"groupsAccesses,omitempty"`
	ID               *string                `json:"id,omitempty"`
	LastConnectionAt *string                `json:"lastConnectionAt,omitempty"`
	Metadata         map[string]interface{} `json:"metadata,omitempty"`
	Type             *string                `json:"type,omitempty"`
	UsersAccesses    map[string]interface{} `json:"usersAccesses,omitempty"`
}

type GetRooms struct {
	Data     []GetRoomsData `json:"data,omitempty"`
	NextPage *string        `json:"nextPage,omitempty"`
}
